### PR TITLE
HARP-4813: Stable zoom level when view is tilted

### PR DESCRIPTION
**Pull request by sasha240100**
See #247

### CODE REVIEW

**Comment by Astrak** [comment]
@heremaps-bot retest this please

**Comment by zbigg** [comment]
@heremaps-bot ok to test

File: @here/harp-mapview/lib/MapView.ts [hunk 1]
```diff
@@ -1717,12 +1718,19 @@ export class MapView extends THREE.EventDispatcher {
         this.m_screenProjector.update(this.m_camera, this.m_worldCenter, width, height);
         this.m_screenCollisions.update(width, height);
 
-        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromHeight(
-            Math.abs(this.m_camera.position.z),
-            this
+        this.m_pixelToWorld = undefined;
+
+        const downQuat = this.m_tempQuat.setFromAxisAngle(EYE_INVERSE, -this.camera.rotation.z);
```
Comment (amsgames94) [inline @here/harp-mapview/lib/MapView.ts]: Why storing the quaternion in the downQuat local variable? Use ``this.m_tempQuat`` afterwards instead.

File: @here/harp-mapview/lib/Utils.ts [hunk 2]
```diff
@@ -202,8 +205,13 @@ export namespace MapViewUtils {
         mapView: MapView,
         zoomLevel: number
     ): number {
+        const downQuat = tempQuat.setFromAxisAngle(EYE_INVERSE, -mapView.camera.rotation.z);
```
Comment (amsgames94) [inline @here/harp-mapview/lib/Utils.ts]: Same as above.

File: @here/harp-mapview/lib/MapView.ts (hunk 1 above)
Comment (zbigg) [reply @here/harp-mapview/lib/MapView.ts]: @amsgames94 
I would argue that this '`downQuat` is more readable and proper name for this particular variable in this context.
Using it as `tmpQuat` doesn't help understanding code at all.

**Pull request merged**